pkg/cli/mirror/mirror.go

```diff
@@ -0,0 +1,179 @@
+package mirror
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	"github.com/RedHatGov/bundle/pkg/cli"
+	"github.com/RedHatGov/bundle/pkg/cli/mirror/describe"
+	"github.com/RedHatGov/bundle/pkg/cli/mirror/list"
+	"github.com/RedHatGov/bundle/pkg/cli/mirror/version"
+)
+
+type MirrorOptions struct {
+	*cli.RootOptions
+	OutputDir       string
+	ConfigPath      string
+	SkipImagePin    bool
+	ManifestsOnly   bool
+	From            string
+	ToMirror        string
+	BuildxPlatforms []string
+}
+
+func NewMirrorCmd() *cobra.Command {
+	o := MirrorOptions{}
+	o.RootOptions = &cli.RootOptions{
+		IOStreams: genericclioptions.IOStreams{
+			In:     os.Stdin,
+			Out:    os.Stdout,
+			ErrOut: os.Stderr,
+		},
+	}
+
+	// Configures a REST client getter factory from configs for mirroring releases.
+	kubeConfigFlags := genericclioptions.NewConfigFlags(true).WithDiscoveryBurst(250)
+	matchVersionKubeConfigFlags := kcmdutil.NewMatchVersionFlags(kubeConfigFlags)
+	f := kcmdutil.NewFactory(matchVersionKubeConfigFlags)
+
+	cmd := &cobra.Command{
+		Use:   filepath.Base(os.Args[0]),
+		Short: "Manages mirrors per user configuration",
+		Long: templates.LongDesc(`
+			oc-mirror will create and publish user configured mirrors with
+            a declarative configation input
+		`),
+		Example: templates.Examples(`
+			# Mirror to a directory
+			oc-mirror --config mirror-config.yaml file://mirror
+			# Mirror to mirror publish
+			oc-mirror --config mirror-config.yaml docker://localhost:5000
+			# Publish a previously created mirror archive
+			oc-mirror --from mirror_seq1_000000.tar docker://localhost:5000
+			# Replay a previous mirror
+			oc-mirror --config mirror-config.yaml --replay=5 file://mirror
+		`),
+		PersistentPreRun:  o.LogfilePreRun,
+		PersistentPostRun: o.LogfilePostRun,
+		Args:              cobra.MinimumNArgs(1),
+		SilenceErrors:     false,
+		SilenceUsage:      false,
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(cmd, f, args))
+			kcmdutil.CheckErr(o.Validate(cmd, f))
+			kcmdutil.CheckErr(o.Run(cmd, f))
+		},
+	}
+
+	fs := cmd.Flags()
+	fs.StringVarP(&o.ConfigPath, "config", "c", o.ConfigPath, "Path to imageset configuration file")
+	fs.BoolVar(&o.SkipImagePin, "skip-image-pin", o.SkipImagePin, "Do not replace image tags with digest pins in operator catalogs")
+	fs.StringVar(&o.From, "from", o.From, "The archive file path.")
```

Comment:
Thoughts on calling this `--archive` to differentiate from the general usage other commands support?

Reply:
I am good with this. @afflom do you agree?

Reply:
I'd prefer `--imageset` in that case. Even if we start handling non-image artifacts, The term `archive` would not support alternative workflows (i.e. using imageset metadata to generate manifests in the absence of an archive)

Reply:
Since my goal was to stick to plain language in the UX, I would like to steer clear of `--imageset` which is why I used `--from` to capture general input from the user. With that context, does `--from` seem like a better fit @estroz? Do you think changing the description would help?

Reply:
+1 to changing description.

Reply:
That's fine with me.